### PR TITLE
use v2ray geosite to replace gfwlist

### PR DESCRIPTION
**Pull request by studentmain**
## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

- [x] [Searched](https://github.com/shadowsocks/shadowsocks-windows/search?q=is%3Apr&type=Issues) for similar pull requests
- [x] Compiled the code with Visual Studio
- [x] Require translation update
- [x] Require document update (readme.md, wikipage, etc)

### What is the purpose of your *pull request*?
- [ ] Bug fix
- [ ] Improvement
- [x] New feature

---

### Description of your *pull request* and other information

Replace GFWList with [v2ray GeoSite database](https://github.com/v2ray/domain-list-community/) in case GFWList finally stopped maintenance (https://github.com/gfwlist/gfwlist/issues/1945).

TODO:

- [x] ~~shared Github release API~~ (Needn't, they has a release branch)
- [x] i18n update
- [x] customizable GeoSite group
- [ ] more stability test
- [ ] how to migrate?
- [x] give more time to GFWList
- [ ] document update
- [ ] announce breaking change

### CODE REVIEW

**Comment by SekiBetu** [comment]
there is no need to “replace” gfwlist, because geosite is a different system, it already contains in shadowsocks, you only need to let shadowsocks to support this .dat file
没必要移除gfwlist，因为ss本身支持白名单模式，就是所谓的全局代理然后切换成绕过大陆及局域网地址，只是默认的白名单没有这么多而已，只要ss能够支持导入这个.dat文件就能切换成白名单模式

**Comment by studentmain** [comment]
> there is no need to “replace” gfwlist, because geosite is a different system, it already contains in shadowsocks, you only need to let shadowsocks to support this .dat file
> 没必要移除gfwlist，因为ss本身支持白名单模式，就是所谓的全局代理然后切换成绕过大陆及局域网地址，只是默认的白名单没有这么多而已，只要ss能够支持导入这个.dat文件就能切换成白名单模式
> 
> 再推荐一个增强规则，集合了很多开源项目
> https://github.com/Loyalsoldier/v2ray-rules-dat

首先ss根本没有白名单模式……

**Comment by SekiBetu** [comment]
> > there is no need to “replace” gfwlist, because geosite is a different system, it already contains in shadowsocks, you only need to let shadowsocks to support this .dat file
> > 没必要移除gfwlist，因为ss本身支持白名单模式，就是所谓的全局代理然后切换成绕过大陆及局域网地址，只是默认的白名单没有这么多而已，只要ss能够支持导入这个.dat文件就能切换成白名单模式
> > 再推荐一个增强规则，集合了很多开源项目
> > https://github.com/Loyalsoldier/v2ray-rules-dat
> 
> 首先ss根本没有白名单模式……

好吧，原来是这个桌面客户端项目没有这功能啊，我的锅，加油，开发起来

File: shadowsocks-csharp/Controller/Service/GeositeUpdater.cs [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+                    valid_lines.Add(line);
+                }
+            }
+            return valid_lines;
+        }
+
+        private static List<string> GeositeToGFWList(IList<DomainObject> domains, bool blacklist)
+        {
+            return blacklist ? GeositeToGFWListBlack(domains) : GeositeToGFWListWhite(domains);
+
```
Comment (Stzx) [inline shadowsocks-csharp/Controller/Service/GeositeUpdater.cs:173]: Let's delete extra line.

**Comment by celeron533** [comment]
@studentmain you can merge PRs if you are confident to do.

**Pull request merged**